db/config.cc

```diff
@@ -155,10 +155,12 @@ struct convert<db::config::seed_provider_type> {
 
 }
 
+static db::extensions default_extensions;
+
 #define str(x)  #x
 #define _mk_init(name, type, deflt, status, desc, ...)  , name(this, str(name), value_status::status, type(deflt), desc)
 
-db::config::config(std::shared_ptr<db::extensions> exts)
+db::config::config(const db::extensions* exts)
     : utils::config_file()
```

Comment:
Why do you need to make extensions a raw pointer? A std::shared is still sharable across shards (as long as no one is allowed to take ownership of it, and even that only if its memory is local). And now you have to add a static (which I know you hate).

Reply:
So that no one takes ownership of it by accident. Strictly speaking, it was illegal even before the patchset, so this particular patch isn't necessary (but I don't think it's wrong).

Statics are bad, but const pointers to statics that are only used when the application doesn't provide its own object are not that bad, the staticness isn't forced on everyone and is just provided as a convenience.

Reply:
@elcallio in earlier versions of the patches, copies of db::config were made, and cross shards too. Now that doesn't happen. I can drop this patch if you prefer, or alternatively afterwards remove db::extensions from db::config entirely.

Reply:
I dropped that patch (the patch that changes shared_ptr<extensions> to shared_ptr<const extensions> remains).

Reply:
Then I have no nits left. Lgtm